packages/client/hmi-client/src/workflow/ops/optimize-ciemss/tera-optimize-ciemss-node.vue

```diff
@@ -18,16 +19,160 @@
+		.setPollAction(async () => pollAction(runId));
+	const pollerResults = await poller.start();
+	let state = _.cloneDeep(props.node.state);
+	state.optimizeErrorMessage = { name: '', value: '', traceback: '' };
+	emit('update-state', state);
+
+	if (pollerResults.state === PollerState.Cancelled) {
+		return pollerResults;
+	}
+	if (pollerResults.state !== PollerState.Done || !pollerResults.data) {
+		// throw if there are any failed runs for now
+		logger.error(`Optimization: ${runId} has failed`, {
+			toastTitle: 'Error - Pyciemss'
+		});
+		const simulation = await getSimulation(runId);
+		if (simulation?.status && simulation?.statusMessage) {
+			state = _.cloneDeep(props.node.state);
+			state.inProgressOptimizeId = '';
+			state.optimizeErrorMessage = {
+				name: runId,
+				value: simulation.status,
+				traceback: simulation.statusMessage
+			};
+			emit('update-state', state);
+		}
+		throw Error('Failed Runs');
+	}
+	return pollerResults;
+};
+
+const getSimulationInterventions = async () => {
+	const policyResult = await getRunResult(props.node.state.optimzationRunId, 'policy.json');
+	const paramNames: string[] = [];
+	const paramValues: number[] = [];
+	const startTime: number[] = [];
+	props.node.state.interventionPolicyGroups.forEach((ele) => {
+		paramNames.push(ele.parameter);
+		paramValues.push(ele.paramValue);
+		startTime.push(ele.startTime);
+	});
+
+	const simulationIntervetions: SimulationIntervention[] = [];
+	// This is all index matching for optimizeInterventions.paramNames, optimizeInterventions.startTimes, and policyResult
+	for (let i = 0; i < paramNames.length; i++) {
+		if (policyResult?.at(i) && startTime?.[i]) {
+			simulationIntervetions.push({
+				name: paramNames[i],
+				timestep: startTime[i],
+				value: policyResult[i]
+			});
+		}
+	}
+	return simulationIntervetions;
+};
+
+const startForecast = async (simulationIntervetions) => {
+	const simulationPayload: SimulationRequest = {
+		projectId: '',
+		modelConfigId: modelConfigId.value as string,
+		timespan: {
+			start: 0,
+			end: props.node.state.endTime
+		},
+		interventions: simulationIntervetions,
+		extra: {
+			num_samples: props.node.state.numSamples,
+			method: props.node.state.solverMethod
+		},
+		engine: 'ciemss'
+	};
+
+	const simulationResponse = await makeForecastJobCiemss(simulationPayload);
+	return simulationResponse;
+};
+
+watch(
+	() => props.node.state.inProgressOptimizeId,
+	async (id) => {
+		if (!id || id === '') return;
+
+		const response = await pollResult(id);
+		if (response.state === PollerState.Done) {
+			// Start 2nd simulation to get sample simulation from dill
+			const simulationIntervetions = getSimulationInterventions();
+			const forecastResponse = await startForecast(simulationIntervetions);
+			const forecastId = forecastResponse.id;
+
+			const state = _.cloneDeep(props.node.state);
+			state.inProgressOptimizeId = '';
+			state.optimzationRunId = id;
+			state.inProgressForecastId = forecastId;
```

Comment:
Is this the only place where `state.inProgressForecastId` gets mutated? If so you maybe don't need another watcher for that. You can just continue to do the simulation within here.

Reply:
I could foresee there being issues if i put that in the watcher for `inProgressOptimizeId` as the initial check will fail when that is now back to empty. If I were to go into a different page and come back type thing

I could add one watcher for both but i dont think thats any cleaner